.github/workflows/test-and-upload-coverage.yml

```diff
@@ -167,7 +167,17 @@ jobs:
         if: ${{ !matrix.detect-changes }}
         uses: actions/upload-artifact@v4
         with:
-          name: ${{ matrix.os }}_${{ matrix.client-type }}_${{ matrix.database-type }}_${{ matrix.mutation-type }}
+          # Make sure the name is always unique per job as artifacts are now immutable.
+          # Note Issue: https://github.com/actions/upload-artifact/issues/478
+          # Solve: https://github.com/actions/upload-artifact/issues/478#issuecomment-1885470013
+          name: "coverage\
+            _${{ matrix.os }}\
+            _${{ matrix.client-type }}\
+            _${{ matrix.database-type }}\
+            _${{ matrix.mutation-type }}\
+            _${{ matrix.lens-type }}\
+            _${{ matrix.database-encryption }}\
```

Comment:
note: don't need the change detection here as we don't generate any coverage reports in that step, also in a follow-up PR will remove the change detection step in it's own workflow so that we no longer have to wait for code-cov uploading to be blocked on the change detection step which takes 5ish minutes more.